### PR TITLE
Jaden/Sign-In

### PR DESCRIPTION
**Pull request by Just-Jaden**
### Trello Link
[Ticket](https://trello.com/c/Zlm9qhpI)

# Description
- Created a Sign-In endpoint 

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Infrastructure

### How Has This Been Tested?

 - Ran locally
 - manually hit route in postman
 
### Checklist:

- [x] I have performed a self-review of my own code

### CODE REVIEW

File: src/routes/auth.js [hunk 1]
```diff
@@ -50,4 +50,44 @@ router.post('/signup', async (request, response, next) => {
...
+    });
+
+    if (!user) {
+      throw new RequestError('Invalid email or password');
+    }
+
+    const passwordMatch = await bcrypt.compare(userPassword, user.password);
+    if (!passwordMatch) {
+      throw new RequestError('Invalid email or password');
+    }
```
[the remark this answers is not in the document]
Comment (Just-Jaden) [reply src/routes/auth.js:78]: Gotcha that makes sense!

File: src/routes/auth.js [hunk 2]
```diff
@@ -50,4 +50,44 @@ router.post('/signup', async (request, response, next) => {
...
+
+    response.json({
+      message: 'Successfully signed in!',
+      user: {
+        id: user.id,
+        name: user.name,
+        email: user.email,
+      },
+    });
```
[the remark this answers is not in the document]
Comment (Just-Jaden) [reply src/routes/auth.js:87]: Gotcha!

File: src/routes/auth.js [hunk 3]
```diff
@@ -50,4 +50,47 @@ router.post('/signup', async (request, response, next) => {
...
+        email: userEmail,
+      },
+    });
+    const passwordMatch = await bcrypt.compare(userPassword, user.password);
+    if (!user && !passwordMatch) {
+      throw new RequestError('Invalid email or password');
+    }
+
+    const accessToken = await tokenService.createAccessToken(user);
+    const refreshToken = await tokenService.getSignedRefreshToken({
```
Comment (KevinHuang0126) [inline src/routes/auth.js:76]: In our `Schema.prisma` file, the user model has a RefreshTokens attribute as a list of RefreshTokens. does the token created on this line get added to the list oh RefreshTokens in the specific user?

**Review by KevinHuang0126 — APPROVED** [review]
Looks good! just one question.

File: src/routes/auth.js (hunk 3 above)
Comment (Just-Jaden) [reply src/routes/auth.js:76]: Yes it does

**Pull request merged**